### PR TITLE
Assure that a list of scopes is always returned regardless of the verb

### LINKED ISSUE #1920: KeyError: 'HEAD'
https://app.getsentry.com/sentry/sentry/group/70051366/

```
KeyError: 'HEAD'
  File "sentry/api/base.py", line 81, in handle_exception
    return super(Endpoint, self).handle_exception(exc)
  File "sentry/api/base.py", line 124, in dispatch
    self.initial(request, *args, **kwargs)
  File "rest_framework/views.py", line 318, in initial
    self.check_permissions(request)
  File "rest_framework/views.py", line 275, in check_permissions
    if not permission.has_permission(request, self):
  File "sentry/api/permissions.py", line 48, in has_permission
    allowed_scopes = set(self.scope_map[request.method])
```

### PR DESCRIPTION
**Pull request by mattrobenolt**
Fixes GH-1920

### CODE REVIEW

**Comment by dcramer** [comment]
We don't want HEAD causing reads which is why we don't support it

**Comment by mattrobenolt** [comment]
> We don't want HEAD causing reads which is why we don't support it

Then should we always fall back to `[]` and not special case `HEAD` ?

**Comment by dcramer** [comment]
Yeah default empty list could be fine

**Comment by mattrobenolt** [comment]
@dcramer, I removed the fallback for HEAD

File: src/sentry/api/permissions.py [hunk 1]
```diff
@@ -55,6 +55,12 @@ def has_object_permission(self, request, view, obj):
     def is_project_key(self, request):
         return isinstance(request.auth, ProjectKey)
 
+    def get_scopes(self, key):
+        try:
+            return self.scope_map[key]
```
Comment (dcramer) [inline src/sentry/api/permissions.py]: probably should just use .get(), and not entirely sure the function call is all that useful

Comment (mattrobenolt) [inline src/sentry/api/permissions.py]: > probably should just use .get(), and not entirely sure the function call is all that useful

You're right. It's useless now. Removed, and switched everything to `scope_map.get(request.method, [])` instead.

**Pull request merged**